src/ui/setup/dynamic/devtools.ts

```diff
@@ -0,0 +1,126 @@
+import { bindActionCreators } from "redux";
+
+import { initSocket } from "protocol/socket";
+import { ThreadFront } from "protocol/thread";
+import { setupGraphics } from "protocol/graphics";
+import { setupLogpoints } from "protocol/logpoint";
+import { updateEnableRepaint } from "protocol/enable-repaint";
+
+import { extendStore } from "../store";
+import app from "ui/reducers/app";
+import timeline from "ui/reducers/timeline";
+import comments from "ui/reducers/comments";
+import { selectors } from "ui/reducers";
+import { createSession } from "ui/actions/session";
+import { actions } from "ui/actions";
+const { setupApp, setupTimeline, setModal } = actions;
+
+import * as dbgClient from "devtools/client/debugger/src/client";
+import debuggerReducers from "devtools/client/debugger/src/reducers";
+import { clientCommands } from "devtools/client/debugger/src/client/commands";
+import { bootstrapWorkers } from "devtools/client/debugger/src/utils/bootstrap";
+import { setupDebuggerHelper } from "devtools/client/debugger/src/utils/dbg";
+import { setupEventListeners } from "devtools/client/debugger/src/actions/event-listeners";
+const { setupExceptions } = require("devtools/client/debugger/src/actions/logExceptions");
+
+import { getConsoleInitialState } from "devtools/client/webconsole/store";
+import { prefs as consolePrefs } from "devtools/client/webconsole/utils/prefs";
+import consoleReducers from "devtools/client/webconsole/reducers";
+import { getPrefsService } from "devtools/client/webconsole/utils/prefs";
+const { setupMessages } = require("devtools/client/webconsole/actions/messages");
+const {
+  initOutputSyntaxHighlighting,
+} = require("devtools/client/webconsole/utils/syntax-highlighted");
+
+import { DevToolsToolbox } from "ui/utils/devtools-toolbox";
+import { asyncStore } from "ui/utils/prefs";
+import { getUserSettings } from "ui/hooks/settings";
+const { LocalizationHelper } = require("devtools/shared/l10n");
+const { setupDemo } = require("ui/utils/demo");
+
+declare global {
+  interface Window {
+    gToolbox: DevToolsToolbox;
+    L10N: any;
+  }
+  interface AppHelpers {
```

Comment:
This interface overlaps with the one defined in helpers.ts as well. Perhaps that's okay right now if the goal is to incrementally sunset these globals but if there are multiple modules that set up / interact with these globals, it might be worth consolidating them. I don't have specific advice in mind yet but we can discuss if you think it'd be useful.

Reply:
I kept the declarations for global variables in the files that set those variables, but I'm open to organizing it differently. Perhaps we should discuss and write up some guidelines about the folder structure of our code (and refactor the code accordingly) - currently I'm often unsure what the "right" place for new code would be.

Reply:
> I kept the declarations for global variables in the files that set those variables

This seems like a good practice when you need to use globals but my preference would be to avoid globals and to import the shared methods directly where they are needed.

In cases where we _want_ a global (like the `app` var that helps expose some stuff in user-land. This approach seems reasonable to me.

My point in this specific case is that we have `AppHelpers` defined twice which TS happily merges into one and I'd favor either bringing them into the same module or breaking them up more clearly so it's evident that `global.app` is fulfilled by two different modules.